tests/fixtures/saas/hubspot_fixtures.py

```diff
@@ -85,3 +95,45 @@ def dataset_config_hubspot(
     )
     yield dataset
     dataset.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def setup_teardown_erasure_hubspot_data(connection_config_hubspot, hubspot_erasure_identity_email) -> Generator:
+    """
+    Gets the current value of the resource and restores it after the test is complete.
+    Used for erasure tests.
+    """
+
+    connector = SaaSConnector(connection_config_hubspot)
+
+    # create contact
+    contacts_request: SaaSRequestParams = SaaSRequestParams(
+        method=HTTPMethod.POST,
+        path=f"/crm/v3/objects/contacts",
+        json_body={
+            "properties": {
+                "company": "test company",
+                "email": hubspot_erasure_identity_email,
+                "firstname": "SomeoneFirstname",
+                "lastname": "SomeoneLastname",
+                "phone": "(123) 123-1234",
+                "website": "someone.net"
+            }
+        },
+    )
+    contacts_response = connector.create_client().send(contacts_request)
+    contacts_body = contacts_response.json()
+    contact_id = contacts_body["id"]
+
+    # no need to subscribe contact, since creating a contact auto-subscribes them
+
+    time.sleep(10)  # Allows contact to be propagated in Hubspot before calling access / erasure requests
```

Comment:
Is there an endpoint that could be called in a while loop or similar that verifies that the contact data is propagated, without relying on an arbitrary wait of 10 seconds?